### PR TITLE
[browser][non-icu] Revert `HybridGlobalization` normalization.

### PR DESCRIPTION
**Pull request by ilonatommy**
Solving https://github.com/dotnet/runtime/issues/86041 in hybrid version is too costly. If we wanted to apply all the rules from https://www.unicode.org/Public/idna/13.0.0/IdnaMappingTable.txt we would use more space than we are saving from removing normalization filter from ICU (~59kB).

IDN working branch: https://github.com/ilonatommy/runtime/tree/idn-mapping.
The implementation is based on enhanced invariant IDN mapping. The problem is that it does not throw when the input is incorrect. In Unicode13 we have ~6k test cases. For function GetAscii()  the code produces ~3.3k cases of false truth when the output is correct and ~729 cases of false truth when the output is incorrect. To fix it, IDN mapping table is needed. Loading mapping table separately and maintaining them with changing Unicode version does not give us size reduction.

From this reason, this PR is reverting normalization changes. Normalization and IDN mapping use the same chunk of ICU data.
Partially reverts the changes from https://github.com/dotnet/runtime/pull/85510 (refactoring changes stay unreverted).

### CODE REVIEW

**Comment by ilonatommy** [comment]
Failures are not connected.

File: src/mono/sample/wasm/browser-bench/String.cs [hunk 1]
```diff
@@ -16,9 +16,6 @@ class StringTask : BenchTask
         public StringTask()
         {
             measurements = new Measurement[] {
-                new NormalizeMeasurement(),
```
Comment (pavelsavara) [inline src/mono/sample/wasm/browser-bench/String.cs:19]: Let's keep the benchmarks anyway

**Review by pavelsavara — APPROVED** [review]

File: src/mono/sample/wasm/browser-bench/String.cs (hunk 1 above)
Comment (pavelsavara) [reply src/mono/sample/wasm/browser-bench/String.cs:19]: Perhaps with smaller buffers, so that they are fast.

Comment (ilonatommy) [reply src/mono/sample/wasm/browser-bench/String.cs:19]: like 8kB?

**Pull request merged**